File: 040020365_Java/picture.java
/**
 * Mehmet CAMBAZ
 * 040020365
 * 
 * Advanced Programming Term Project
 * Java : picture.java
 */

 /**
  * picture class
  * represents a picture
  * contains number of rows (height)
  * 		 number of columns (width)
  * 		 picture content (pixels)
  * */ 

public class picture 
{
	public char pixels[][];		//picture content
	private int rows;			//number of rows (height)
	private int cols;			//number of columns (width)
	
	//constructor
	picture()
	{
		//means nothing assigned
		cols = 0;
		rows = 0;
		pixels = new char[100][100];	//default picture size is 100*100
	}

	
	//given height and width constructor
	picture(int row, int col)
	{
		//means nothing assigned
		cols = col;
		rows = row;
		pixels = new char[row][col];
	}
	
	
	//copy constructor
	picture(picture right)
	{
		cols = right.cols;
		rows = right.rows;
		
		for(int i=0; i<right.getRows(); i++)
			pixels[i] = right.pixels[i];
	}
	
		
	public int getCols()
	{
		return cols;
	}

		
	public int getRows()
	{
		return rows;
	}
	
		
	public void setCols(int col)
	{
		cols = col;
	}
	
		
	public void setRows(int row)
	{
		rows = row;
	}
	
	
	/**
	 * prints the picture content to user
	 */
	public void print()
	{
	     for(int i=0; i<getRows(); i++)
	     {
		     for(int j=0; j<getCols(); j++)
		     	 System.out.print(pixels[i][j]);
		    
		     System.out.print("\n");
	     }
	}
	
}

File: 040020365_python/picture.py
#
# Mehmet CAMBAZ
# 040020365
#
# Advanced Programming Term Project
# Python : picture.py
#

##############################
# picture class              #
# used to represent pictures #
##############################
class picture:
    rows = 0        #number of rows in the picture (height)
    cols = 0        #number of columns in the picture (width)
    pixels = []     #picture pixel content

    #constructor
    def __init__(self, row, col, filename=""):
        self.rows = row
        self.cols = col
        self.pixels = []        
        
        if filename != "":
            f = open(filename, 'r')        
            
            for i in range(row):
                column = []
                l = f.readline()
                for j in range(col):
                    column.append(l[j])    
                self.pixels.append(column)  
            
            f.close()

    #prints picture content    
    def printPixels(self):
        for i in range(self.rows):
            for j in range(self.cols):
                print self.pixels[i][j],
            print '\n',
                
